spec/models/room_spec.rb

```diff
@@ -29,6 +29,45 @@
     end
   end
 
+  describe '#anyone_joins_as_moderator?' do
+    let!(:room) { create(:room) }
+
+    let(:fake_room_meeting_option) do
+      Class.new do
```

Comment:
That may look complex but it was the simplest way I could come up with.
The problem is due to the `anyone_joins_as_moderator?` expecting an object with a value attribute (property).
Mocking the result as simple hash will not doing it because of `Hash` nature in ruby, their properties are not a class attributes that is why we're accessing them through the the brackets `[]`.
And not mocking the `MeetingOption::find_room_meeting_option_by_name` and creating the records on the database will solve this but will make this an integration test rather then a unit test since it will further test the `find_room_meeting_option_by_name` which has its own unit tests.
So, I created a fake anonymous class that has the expected attributes mocking the result of calling `find_room_meeting_option_by_name`.

Reply:
You can just mock the request with `OpenStruct`

`allow(MeetingOption).to receive(:get_value).and_return(OpenStruct.new(value: 'true'))`

Reply:
> You can just mock the request with `OpenStruct`
> 
> `allow(MeetingOption).to receive(:get_value).and_return(OpenStruct.new({value: 'true'}))`

That's exactly what I needed, didn't know about the `OpenStruct` class.
Thanks for sharing 🙏!
> For anyone that wants to discover this here's an article that I found worthy [HERE](https://medium.com/rubycademy/openstruct-in-ruby-ab6ba3aff9a4).

Reply:
Rubocop doesn't allow the user of `OpenStructs` and favorizes the use of much stricter structures such as `Structs`, `Hashes` and `Doubles` [CHECK HERE](https://docs.rubocop.org/rubocop/cops_style.html#styleopenstructuse).
Therefore, switched to `instance_doubles` [MORE HERE](https://relishapp.com/rspec/rspec-mocks/v/3-11/docs/verifying-doubles/using-an-instance-double).